src/Proto3/Suite/DotProto/Parsing.hs

```diff
@@ -225,9 +234,12 @@ package = do symbol "package"
              return $ DotProtoPackageSpec p
 
 definition :: ProtoParser DotProtoDefinition
-definition = message
-         <|> enum
-         <|> service
+definition = 
+  choice 
+    [ try message
```

Comment:
Why do we need `try` added here? I believe `choice` is just a fold with `<|>` as its binary function so shouldn't `choice` with no `try` be equivalent to the prior formulation?

Reply:
The choice between `extend` (on line 215) and `enum` requires backtracking since lookahead is greater than 1.